### PR TITLE
storage: fix memory leak when preloading files

### PR DESCRIPTION
**Pull request by colindickson**

### CODE REVIEW

File: storage/execout/filewalk.go [hunk 1]
```diff
@@ -71,19 +91,53 @@ func (fw *FileWalker) preload(ctx context.Context, seg int) {
...
+	fw.currentlyPreloadingLock.Unlock()
+
 	f := fw.config.NewFile(rng)
-	go func() {
-		if err := f.Load(ctx); err == nil {
+	fw.buffer[seg] = f
+
+	go func(file *File) {
+		defer func() {
+			fw.currentlyPreloadingLock.Lock()
```
Comment (maoueh) [inline storage/execout/filewalk.go:103]: Do we want a read lock here?

File: storage/execout/filewalk.go [hunk 2]
```diff
@@ -71,19 +91,53 @@ func (fw *FileWalker) preload(ctx context.Context, seg int) {
...
+			fw.bufferLock.Lock()
+			fw.currentlyPreloadingLock.Lock()
+
+			delete(fw.buffer, k)
+			delete(fw.currentlyPreloadingSegments, k)
+
+			fw.currentlyPreloadingLock.Unlock()
+			fw.bufferLock.Unlock()
+		}
+	}
```
Comment (maoueh) [inline storage/execout/filewalk.go:140]: There is a chance that a key has been added/deleted between those two `fw.bufferLock.Lock()`, what would it cause? 

Should we step lock the full thing here?

**Comment by maoueh** [comment]
This PR seems abandoned, shall I kill it?

**Comment by colindickson** [comment]
yes @maoueh .  it turns out the memory leak was coming from elsewhere.

**Pull request closed without merging**